Star Practices/src/textSearch/FileReaderClass.java

```diff
@@ -0,0 +1,64 @@
+package textSearch;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class FileReaderClass {
+
+    private List<Pair<String,Integer>> words = new ArrayList<Pair<String,Integer>>();
+
+    private List<File> m_files;
+
+
+    private void readFile(File file, int index) {
+        // Reads a file and adds the tokens to the list
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(file));
+            for (String line = reader.readLine(); line != null; line = reader.readLine()) {
+                for (String _word : line.split("\\W+")) {
+                    words.add(new Pair<String,Integer>(_word, index));
+                }
+            }
+            reader.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public FileReaderClass(File directory) {
```

Comment:
FileReaderClass is not an appropriate name for a method.

Reply:
the constructor should always have the same name as the class